### PR TITLE
add jsonb functions

### LINKED ISSUE #962: Document jsonb table functions
### Related code PR

https://github.com/risingwavelabs/risingwave/pull/9977

### Which part(s) of the docs might be affected or should be updated? And how?

Add these functions: 

jsonb_array_elements ( jsonb ) → setof jsonb
jsonb_array_elements_text ( jsonb ) → setof text
jsonb_each ( jsonb ) → setof record ( key text, value jsonb )
jsonb_each_text ( jsonb ) → setof record ( key text, value text )
jsonb_object_keys ( jsonb ) → setof text

### Reference

https://www.postgresql.org/docs/15/functions-json.html

https://github.com/risingwavelabs/risingwave/issues/11570

### PR DESCRIPTION
**Pull request by hengm3467**
## Info

- **Description**

Add JSONB functions

- **Notes**

  [ Include any supplementary context or references here. ]

- **Related code PR**

  [ Provide a link to the relevant code PR here, if applicable. ]

- **Related doc issue**
  
  Resolves #962

## For reviewers

- **Preview**

  [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **upcoming** version of the documentation. ]

- **Key points**

  [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

File: docs/sql/functions-operators/sql-function-json.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+### `jsonb_each_text`
+
+Expands the top-level JSON object into a set of key/value pairs. The returned values will be of type varchar.
+
+```bash title=Syntax
+jsonb_each_text ( jsonb ) → setof record ( key varchar, value varchar )
+```
+
+```sql title=Example
+SELECT jsonb_each_text('{"a":"foo", "b":"bar"}'::jsonb);
```
Comment (xiangjinwu) [inline docs/sql/functions-operators/sql-function-json.md]: What about also using `select * from` to be aligned with other examples? The result would be similar to `jsonb_each` above but without quotes:
```
a foo
b bar
```

**Review by xiangjinwu — APPROVED** [review]
We also have some operators and functions on this page:
https://docs.risingwave.com/docs/current/data-type-jsonb/
```
JSONB -> int → JSONB
JSONB -> varchar → JSONB
JSONB ->> int → varchar
JSONB ->> varchar → varchar
jsonb_typeof(jsonb) → varchar
jsonb_array_length(jsonb) → int
```

rest lgtm

File: docs/sql/functions-operators/sql-function-json.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+```sql title=Example
+SELECT * FROM jsonb_array_elements_text('["foo", "bar"]'::jsonb)
+------RESULT
+foo
+bar
+```
+
+### `jsonb_each`
+
+Expands the top-level JSON object into a set of key/value pairs.
```
Comment (emile-00) [inline docs/sql/functions-operators/sql-function-json.md]: I usually see it formatted as "key-value"

**Review by emile-00 — APPROVED** [review]

**Comment by hengm3467** [comment]
Moved the operators and functions from the JSONB data type page to the JSON functions and operators page.

**Pull request merged**